Detector/DetFCChhECalSimple/src/ECalBarrel_geo.cpp

```diff
@@ -48,33 +47,42 @@ static DD4hep::Geometry::Ref_t createECal (DD4hep::Geometry::LCDD& lcdd,xml_h xm
   std::string passive_mat=passive.materialStr();
   double passive_tck=passive.thickness();
 
-  // Step 1 : cryostat
-  DetElement cryo(cryostat.nameStr(), 0);
-  DD4hep::Geometry::Tube cryoShape(cryo_dims.rmin() , cryo_dims.rmax(), cryo_dims.dz());
-  lLog << MSG::DEBUG << "ECAL Building cryostat from " << cryo_dims.rmin() << " to " << cryo_dims.rmax() << endmsg;
-  Volume cryoVol(cryostat.nameStr(), cryoShape, lcdd.material(cryostat.materialStr()));
-  PlacedVolume placedCryo = envelopeVolume.placeVolume(cryoVol);
-  placedCryo.addPhysVolID("ECAL_Cryo", cryostat.id());
-  cryo.setPlacement(placedCryo);
-
-  // Step 2 : fill cryostat with active medium
-  DetElement calo_bath(active_mat, 0);
-  DD4hep::Geometry::Tube bathShape(cryo_dims.rmin()+cryo_thickness , cryo_dims.rmax()-cryo_thickness, cryo_dims.dz()-cryo_thickness);
-  lLog << MSG::DEBUG << "ECAL: Filling cryostat with active medium from " << cryo_dims.rmin()+cryo_thickness << " to " << cryo_dims.rmax()-cryo_thickness << endmsg;
-  Volume bathVol(active_mat+"_notSensitive", bathShape, lcdd.material(active_mat));
-  PlacedVolume placedBath = cryoVol.placeVolume(bathVol);
-  placedBath.addPhysVolID("bath", 1);
-  calo_bath.setPlacement(placedBath);
+  //cryo_thickness < 0 => Run without a cryostat
+  Volume bathVol; //volume inside cryostat filled with active medium
+  if (cryo_thickness>0) {
+    // Step 1 : cryostat
+    DetElement cryo(cryostat.nameStr(), 0);
+    DD4hep::Geometry::Tube cryoShape(cryo_dims.rmin() , cryo_dims.rmax(), cryo_dims.dz());
+    lLog << MSG::DEBUG << "ECAL cryostat: rmin " << cryo_dims.rmin() << " rmax " << cryo_dims.rmax() << endmsg;
+    Volume cryoVol(cryostat.nameStr(), cryoShape, lcdd.material(cryostat.materialStr()));
+    PlacedVolume placedCryo = envelopeVolume.placeVolume(cryoVol);
+    placedCryo.addPhysVolID("ECAL_Cryo", 1);
+    cryo.setPlacement(placedCryo);
+    // Step 2 : fill cryostat with active medium
+    DetElement calo_bath(active_mat+"_notSensitive", 0);
+    DD4hep::Geometry::Tube bathShape(cryo_dims.rmin()+cryo_thickness , cryo_dims.rmax()-cryo_thickness, cryo_dims.dz()-cryo_thickness);
+    lLog << MSG::DEBUG << "ECAL " << active_mat << " bath: rmin " << cryo_dims.rmin()+cryo_thickness << " rmax " << cryo_dims.rmax()-cryo_thickness << endmsg;
+    bathVol = Volume(active_mat+"_notSensitive", bathShape, lcdd.material(active_mat));
+    PlacedVolume placedBath = cryoVol.placeVolume(bathVol);
+    placedBath.addPhysVolID("bath", 1);
+    calo_bath.setPlacement(placedBath);
+  }
 
   // Step 3 : create the actual calorimeter
   int active_samples= (calo_dims.rmax()-  calo_dims.rmin() - passive_tck)/(passive_tck+active_tck);
   double calo_tck=active_samples*(active_tck+passive_tck)+passive_tck;
-  DetElement caloDet(calo_name, calo_id);
+  DetElement caloDet(calo_name, 1);
```

Comment:
1/ This is not the whole ECAL volume, but one of the volumes inside (where the individual layers are placed).
2/ This is not the volume ID, but DetElement ID which is not needed later. 
3/ The DetElements are probably not needed to be created for each volume. We will try to remove it if possible in the future.